examples/allennlp/subsample_dataset_reader.py

```diff
@@ -0,0 +1,20 @@
+import itertools
+
+import allennlp
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+
+
+@DatasetReader.register("subsample")
+class SubsampleDatasetReader(allennlp.data.dataset_readers.TextClassificationJsonReader):
+    def __init__(self, train_data_size, validation_data_size, **kwargs):
+        super().__init__(**kwargs)
+        self.train_data_size = train_data_size
+        self.validation_data_size = validation_data_size
+
+    def _read(self, datapath):
+        if datapath.endswith("train.jsonl"):
+            data_size = self.train_data_size
+        else:
+            data_size = self.validation_data_size
+
+        yield from itertools.islice(super()._read(datapath), data_size)
```

Comment:
While the previous `SubsampledDatasetReader` shuffles data before yielding, the current `SubsampleDatasetReader` does not do so. Is it intended?

Reply:
Yes, it's intended for taking advantage of a generator. 🙋‍♂️ 

I take a quick look at the dataset to find that it seems to be shuffled in advance.
(I put the screenshot of `train.jsonl` at the bottom of this comment.)
Additionally, the numbers of positive/negative labels are almost the same by reading data sequentially from the top. (https://github.com/optuna/optuna/pull/1542#discussion_r458805693)
Then, I think I could remove shuffling as the preprocessing.

<img width="100" alt="train.jsonl" src="https://user-images.githubusercontent.com/5164000/88933446-63e62a80-d2ba-11ea-82d6-e568c5269419.png">

Reply:
Thanks for the reply. I understood what you intended. Let me resolve this conversation.